core/src/main/scala/kafka/log/LogValidator.scala

```diff
@@ -93,33 +97,45 @@ private[kafka] object LogValidator extends Logging {
     batch
   }
 
-  private def validateBatch(firstBatch: RecordBatch, batch: RecordBatch, isFromClient: Boolean, toMagic: Byte): Unit = {
+  private def validateBatch(topicPartition: TopicPartition, firstBatch: RecordBatch, batch: RecordBatch, isFromClient: Boolean, toMagic: Byte, brokerTopicStats: BrokerTopicStats): Unit = {
     // batch magic byte should have the same magic as the first batch
-    if (firstBatch.magic() != batch.magic())
+    if (firstBatch.magic() != batch.magic()) {
+      brokerTopicStats.allTopicsStats.invalidMagicNumberRecordsPerSec.mark()
```

Comment:
In line 104,, 112, 119, 124 and 131 137 152 and 461 we should also include the topic-partition information.

Reply:
ack